File: problems/824/solution.go
package main
func toGoatLatin(S string) string {
    vowels := []rune{ 'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U' }
    words := []string{}
    
    for i, word := range strings.Split(S, " ") {
        if !contains(vowels, rune(word[0])) {
            word = string(word[1:]) + string(word[0])
        }
        
        words = append(words, word + "ma" + strings.Repeat("a", i + 1))
    }
    
    return strings.Join(words, " ")
}

func contains(list []rune, el rune) bool {
    for _, v := range list {
        if v == el {
            return true
        }
    }
    
    return false;
}

File: problems/824/solution.py
class Solution:
    def toGoatLatin(self, S: str) -> str:
        if not S:
            return S
        
        vowels, words = set(['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']), S.split()
        end = 'maa'
        
        for i, word in enumerate(words):
            if word[0] not in vowels:
                words[i] = word[1:] + word[0]
            
            words[i] += end
            end += 'a'
        
        return ' '.join(words)